LeGO-LOAM/src/featureAssociation.cpp

```diff
@@ -180,13 +180,37 @@ class FeatureAssociation{
     cv::Mat matP;
 
     int frameCount;
+    int imuQueLength;
+    float surfThreshold;
+    int nearestFeatureSearchSqDist;
+    float edgeThreshold;
+    int N_SCAN ;
+    int Horizon_SCAN ;
+    float ang_res_x ;
+    float ang_res_y ;
+    float ang_bottom ;
+    int groundScanInd ;
 
 public:
 
     FeatureAssociation():
         nh("~")
         {
-
+        nh.getParam("scanPeriod", scanPeriod);
```

Comment:
I don't think that it is a good idea to gather ROS parameters in the constructor of a class. At some point, the community might need to move from ROS 1 to ROS 2, and this would make it more difficult. Could you maybe put all the parameter fetching block in a method getParametersFromRos (or similar)?